server/src/main/java/com/datastrato/gravitino/server/web/rest/MetalakeOperations.java

```diff
@@ -66,7 +76,30 @@ public Response listMetalakes() {
           httpRequest,
           () -> {
             Metalake[] metalakes =
-                TreeLockUtils.doWithRootTreeLock(LockType.READ, metalakeDispatcher::listMetalakes);
+                TreeLockUtils.doWithRootTreeLock(
+                    LockType.READ,
+                    () -> {
+                      if (AuthorizationUtils.enableAuthorization()) {
+                        String currentUser = PrincipalUtils.getCurrentPrincipal().getName();
+
+                        AuthorizationUtils.checkIsMetalakeAdmin();
+
+                        Metalake[] orignMetalakes = metalakeDispatcher.listMetalakes();
+
+                        // Filter the metalakes which aren't created by the user
+                        List<Metalake> filteredMetalakes = Lists.newArrayList();
+                        for (Metalake metalake : orignMetalakes) {
+                          if (metalake.auditInfo().creator().equals(currentUser)) {
```

Comment:
Also here, why only the creator has permission to list metalakes, what if the creator is leaving, how to handle this scenario?